extensions/sql-database-projects/src/models/project.ts

```diff
@@ -973,11 +972,34 @@ export class Project implements ISqlProject {
 		const databaseReferenceEntry = new DacpacReferenceProjectEntry(settings);
 
 		// check if reference to this database already exists
+		// if it does, throw an error that will get displayed to the user
 		if (this.databaseReferenceExists(databaseReferenceEntry)) {
 			throw new Error(constants.databaseReferenceAlreadyExists);
 		}
 
-		await this.addToProjFile(databaseReferenceEntry);
+		// create database variable
+		if (settings.databaseVariable && settings.databaseName) {
+			await this.sqlProjService.addSqlCmdVariable(this.projectFilePath, settings.databaseVariable, settings.databaseName);
+
+			// create server variable - only can be set when there's also a database variable (reference to different database on different server)
+			if (settings.serverVariable && settings.serverName) {
+				await this.sqlProjService.addSqlCmdVariable(this.projectFilePath, settings.serverVariable, settings.serverName);
+			}
+		}
+
+		let result;
+
+		// dacpac reference uses database sqlcmd variable and possibly server sqlcmd variable
+		if (settings.databaseVariable) {
+			result = await this.sqlProjService.addDacpacReference(this.projectFilePath, settings.dacpacFileLocation.fsPath, settings.suppressMissingDependenciesErrors, settings.databaseVariable, settings.serverVariable);
+		} else {
+			// dacpac reference is to same db or uses database literal
+			result = await this.sqlProjService.addDacpacReference(this.projectFilePath, settings.dacpacFileLocation.fsPath, settings.suppressMissingDependenciesErrors, undefined, undefined, settings.databaseName);
+		}
```

Comment:
can these be combined into one call, just with the irrelevant settings being `undefined`?

```
const databaseLiteral = settings.databaseVariable ? undefined : settings.databaseName;
addDacpacReference(this.projectFilePath, settings.dacpacFileLocation.fsPath, settings.suppressMissingDependenciesError, settings.databaseVariable, setting.serverVariable, databaseLiteral)
```